src/Package/Impl/Options/R/Tools/RToolsSettingsImplementation.cs

```diff
@@ -64,7 +66,15 @@ internal sealed class RToolsSettingsImplementation : IRToolsSettings {
                 _cranMirror = value;
                 // Setting mirror reques running code in R host
                 // which async and cannot be done correctly here.
-                IdleTimeAction.Create(async () => await SetMirrorToSession(), 20, typeof(RToolsSettingsImplementation));
+                IdleTimeAction.Create(async () => await SetMirrorToSession(), 20, _cranMirror ?? (object)this.GetType());
+            }
+        }
+
+        public string RLocale {
+            get { return _locale; }
+            set {
+                _locale = value;
+                IdleTimeAction.Create(async () => await SetSessionLocale(), 20, _locale ?? (object)typeof(CultureInfo));
```

Comment:
`SetSessionLocale` runs on background thread, so I don't think we need to wait for idle.